### DISCUSSION TITLE
Basic Kernel Related Issue

### DISCUSSION THREAD

**Discussion by Shuvro129** [idaholab/moose#22863] [Q&A Modules: General] [answered]
−𝐊 (𝝏^𝟐 𝑼)/(𝝏𝒙^𝟐 )+𝑹𝑼+𝑴 (𝝏^𝟐 𝑼)/(𝝏𝒕^𝟐 )= 𝜹(𝒙,𝒕)
I want solve this equation. I am bit confused about how I am gonna write the kernel for the inertia term for this equation.

**Comment by GiudGiud** [comment, marked as answer]
Hello

you want to solve a wave equation?
We ve discussed this a bit on this forum last year you may want to search for it. 

you ll want to look at the coupleable interface documentation for each derivative term
The right hand side will be a dirackernel btw

Guillaume

**Reply by Shuvro129** [reply]
Hey. Thank you so much. I am searching for it. I am kind of new to moose. Hope I will find something. Will let you know about that.

**Reply by Shuvro129** [reply]
Hi, do you know anything of applying absorbing boundary condition kernel in moose? I was trying to solve the above equation from negative infinity to positive infinity domain?

**Reply by GiudGiud** [reply]
is this just a flux? Then it would be a NeumannBC.
If the flux is proportional to the value of the quantity near the boundary, it would be a coupled NeumannBC
https://mooseframework.inl.gov/source/bcs/CoupledVarNeumannBC.html

**Reply by Shuvro129** [reply]
I mean the field variable would be zero in infinite length. There is nothing mentioned about flux. Will CoupledVarNeumannBC do the trick?

**Reply by GiudGiud** [reply]
if you are able to backtrack this 0 at infinite length to a flux at a given distance, then they should be equivalent.

**Reply by Shuvro129** [reply]
may be there is a way. But for problem, I don't think I can. I am searching for a BC like https://mooseframework.inl.gov/moose/source/bcs/VectorTransientAbsorbingBC.html#! But this is not the one I want.

**Comment by hugary1995** [comment]
How did you type this equation? Looks nice.

**Reply by hugary1995** [reply]
Oh, and regarding your actual question... Don't think about the discretized matrix form which was usually taught at school in the old days. That's not the level MOOSE works with. In MOOSE, you directly compose the equation in its weak form. For inertia, that's as simple as \phi \rho \partial^2 u / \partial t^2, where \phi is the test function, \rho is the density, and u is the displacement.

**Reply by Shuvro129** [reply]
Thank you so much for your response. I just copied the equation from a word file. Do you have any worked-out examples for this type of wave function? If you can, will you kindly paste any links for me? I will search for it as well. I just started working on moose so I don't know much about it so I am kind of finding it hard. Thank you.

**Reply by hugary1995** [reply]
Note that the `U` in your equation could mean anything... But based on the way you wrote the equation, I guess you are looking for dynamics in the context of solid mechanics. If so, this one should be a good starting point: https://mooseframework.inl.gov/modules/tensor_mechanics/Dynamics.html

**Reply by Shuvro129** [reply]
Thanks.

**Reply by Shuvro129** [reply]
Hi, do you know anything of applying absorbing boundary condition kernel in moose? I was trying to solve the above equation from negative infinity to positive infinity domain?

**Reply by Shuvro129** [reply]
At infinite length the value of the field variable would be zero.

**Reply by lynnmunday** [reply]
@cbolisetti do you have an absorbing boundary condition in Mastodon?  Some examples for solving an acoustics problem are given in the directory:  
moose/modules/fsi/test/tests/fsi_acoustics/
These examples don't use an absorbing boundary conditions like perfectly matched layers, you'll need to implement that yourself.  But for a 1-D problem, that should be easy using one of the existing bcs.

**Reply by cbolisetti** [reply]
We have absorbing boundaries for mechanical waves in solids. Check out [NonReflectingBC](https://mooseframework.inl.gov/mastodon/source/bcs/NonReflectingBC.html) in MASTODON. Note that this BC works best when the waves are perpendicular to the absorption surface. The efficiency drops as the waves become parallel to the surface. PML would definitely be an upgrade both for solids and fluid (acoustic) media, but we don't currently have it.

**Reply by Shuvro129** [reply]
Thanks
